app/src/main/java/com/example/bigowlapp/activity/FingerprintAuthenticationActivity.java

```diff
@@ -0,0 +1,203 @@
+package com.example.bigowlapp.activity;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.telephony.TelephonyManager;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.biometric.BiometricManager;
+import androidx.biometric.BiometricPrompt;
+import androidx.core.content.ContextCompat;
+import androidx.lifecycle.ViewModelProvider;
+
+import com.example.bigowlapp.R;
+import com.example.bigowlapp.model.LiveDataWithStatus;
+import com.example.bigowlapp.model.User;
+import com.example.bigowlapp.utils.PhoneNumberFormatter;
+import com.example.bigowlapp.viewModel.HomePageViewModel;
+import com.google.i18n.phonenumbers.NumberParseException;
+
+import java.util.concurrent.Executor;
+
+public class FingerprintAuthenticationActivity extends AppCompatActivity {
+
+    private TextView AuthResultText;
+    private TextView fingerprintAuthRegistrationText;
+    private BiometricPrompt biometricPrompt;
+    private BiometricPrompt.PromptInfo promptInfo;
+    private Button btnStartFingerAuth;
+    private Button btnGoToHomePage;
+    private Button btnFingerprintAuthAdd;
+    private Button btnFingerprintAuthMaybeLater;
+    private HomePageViewModel homePageViewModel;
+    private static final String PROPOSE_FINGERPRINT_AUTH = "Add fingerprint authentication to secure your account?\n\n" +
+            "NOTE: This will also make your account ONLY accessible by a device having sim card_1 phone number same " +
+            "as the phone number you saved in your account";
+    private static final String IS_FINGERPRINT_AUTH_REGISTERED = "You are registered to fingerprint authentication.\n\n" +
+            "You can modify your choice on Edit profile";
+    private static final String NO_COMPATIBILITY_WITH_PHONE = "Sorry, this additional security service is not available on" +
+            " this number or with your phone and telephony provider\n\n" +
+            " First make sure sim card_1 phone number on this phone is the same as in your account";
+    private static final String NOT_ALLOWED = "You are not allowed to access this account\n\n" +
+            "Sim card_1 phone number on this phone should be the same as in your account";
```

Comment:
Personally, I am not a fan of this structure of code. I will probably put it in a constants/string file so it is less messy for this activity class. As this is a lot of string in which is might be unreadable if we have to debug the code. So, it would be best to separate that into a different file.

Reply:
Strings.xml is the best approach to something like this.

Reply:
ok, I have now used those strings in strings.xml